testing/log/service.go

```diff
@@ -355,12 +355,18 @@ func (s *service) listen(context *endly.Context, request *ListenRequest) (*Liste
 			return nil, fmt.Errorf("listener has been already register for %v", logType.Name)
 		}
 	}
-	service, err := storage.NewServiceForURL(source.URL, source.Credentials)
+
+	fs, err := estorage.StorageService(context, source)
+	if err != nil {
+		return nil, err
+	}
+	source, storageOpts, err := estorage.GetResourceWithOptions(context, source)
+	err = fs.Init(context.Background(), source.URL, storageOpts...)
 	if err != nil {
 		return nil, err
 	}
-	defer service.Close()
-	logTypeMetas, err := s.readLogFiles(context, service, source, request.Types...)
+
+	logTypeMetas, err := s.readLogFiles(context, fs, source, request.Types...)
```

Comment:
`logTypeMetas, err := s.readLogFiles(context, fs, source, request.Types...)` can be change to 
`logTypeMetas, err = s.readLogFiles(context, fs, source, request.Types...)` use already `err` created before.